strawberry/cli/commands/codegen.py

```diff
@@ -1,19 +1,21 @@
 from __future__ import annotations
 
+import functools
 import importlib
 import inspect
-from pathlib import Path  # noqa: TCH003
 from typing import TYPE_CHECKING, List, Optional, Type
 
 import rich
 import typer
 
 from strawberry.cli.app import app
 from strawberry.cli.utils import load_schema
-from strawberry.codegen import QueryCodegen, QueryCodegenPlugin
+from strawberry.codegen import ConsolePlugin, QueryCodegen, QueryCodegenPlugin
+from strawberry.codegen.plugins.python import PythonPlugin
+from strawberry.codegen.plugins.typescript import TypeScriptPlugin
 
 if TYPE_CHECKING:
-    from strawberry.codegen import CodegenResult
+    from pathlib import Path
```

Comment:
AHH!!!

What tool is actually moving this import?

In this case, the tool is _thinking_ that `Path` should be in an `if TYPE_CHECKING` because it's only used in annotations, however, `typer` is introspecting the types so the import statement needs to be present at runtime.

I'm not sure how to prevent our tool from moving this (other than putting in some sort of useless expression that actually _uses_ `Path` somewhere).

Reply:
that's ruff, I think that rule is becoming quite annoying, `# noqa: TCH001` should fix it